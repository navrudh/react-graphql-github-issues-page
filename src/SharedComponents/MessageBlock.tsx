import * as React from 'react';

import styled, { StyledComponentClass } from 'styled-components';

const wrapperClasses = (msgType: 'warning' | 'error') => {

  if (msgType === 'warning') {
    return 'bg-light-yellow b--yellow';
  } else if (msgType === 'error') {
    return 'bg-washed-red b--red';
  }

  return 'bg-washed-blue b--black-10';
};

const Title = styled.h1.attrs({
  className: 'tc georgia f4 fw4 mt0 mb0 black-60'
})``;

const Wrapper = (msgType: 'warning' | 'error') =>
  styled.article.attrs({
    className:
      'w-80 w-ns-100 center br3 pa3 pa4-ns mv3 ba' +
      ` ${wrapperClasses(msgType)}`
  })``;

class MessageBlock extends React.Component<{
  content: string;
  msgType: 'warning' | 'error';
}> {
  private content: string;
  // tslint:disable-next-line:no-any
  private _Wrapper: StyledComponentClass<any, any, any>;

  constructor(prop: { content: string; msgType: 'warning' | 'error' }) {
    super(prop);
    this.content = prop.content;
    this._Wrapper = Wrapper(prop.msgType);
  }

  public render() {
    return (
      <this._Wrapper>
        <Title> {this.content} </Title>
      </this._Wrapper>
    );
  }
}

export default MessageBlock;
