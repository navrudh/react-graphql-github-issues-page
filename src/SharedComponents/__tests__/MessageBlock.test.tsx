import * as React from 'react';

import MessageBlock from '../MessageBlock';
import { shallow } from 'enzyme';

describe('MessageBlock', () => {
  it('with warning message renders correctly', () => {
    const wrapper = shallow(<MessageBlock msgType={'warning'} content={'Warning Text'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('with error message renders correctly', () => {
    const wrapper = shallow(<MessageBlock msgType={'error'} content={'Error Text'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
