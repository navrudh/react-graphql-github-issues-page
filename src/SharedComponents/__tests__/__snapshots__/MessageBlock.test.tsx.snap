// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBlock with error message renders correctly 1`] = `
<styled.article>
  <styled.h1>
     
    Error Text
     
  </styled.h1>
</styled.article>
`;

exports[`MessageBlock with warning message renders correctly 1`] = `
<styled.article>
  <styled.h1>
     
    Warning Text
     
  </styled.h1>
</styled.article>
`;
